Mastodon.js

```diff
@@ -43,21 +43,19 @@ function doWeb(doc, url) {
 	var resourceType = detectWeb(doc, url);
 	var newItem = new Zotero.Item(resourceType);
 
-	var tmpDate, tmpAuthor;
+	var tmpAuthor;
 	if (resourceType == "blogPost") {
 		newItem.blogTitle = ZU.xpathText(doc, "(//span[@class='display-name__account'])[1]");
 		newItem.title = ZU.xpathText(doc, "//meta[@name='description']/@content");
-		tmpDate = ZU.xpathText(doc, '//time[@class="formatted"]/@datetime');
-		tmpAuthor = ZU.xpathText(doc, "(//strong[@class='display-name__html p-name emojify'])[1]");
+		var tmpDate = ZU.xpathText(doc, '(//time)[1]/@datetime');
+		newItem.date = ZU.strToISO(tmpDate);
+		tmpAuthor = ZU.xpathText(doc, "//meta[@property='og:title']/@content").split("(")[0];
```

Comment:
1) Won't this leave the closing parenthesis? If you know there's going to be specific punctuation at the end, you should remove it manually rather than just relying on `cleanAuthor()` below.

2) I specifically said that `og:title` isn't appropriate for the author name, since they're misusing it — the title of the page for an individual post shouldn't be the author's name.

Reply:
No, since it's a `split` and only the first part of the `og:title` tag is grabbed. I can also grab it using a regexp if that's read better. Regarding the selection, I have added the space before the parenthesis to avoid relying on `cleanAuthor` to remove the trailing space. Some sample code of the grabbed content can be found below.

```tmp = "Félix Brezo (@febrezo@mastodon.social)";
"Félix Brezo (@febrezo@mastodon.social)"
tmp.split("(")[0];
"Félix Brezo "
tmp.split(" (")[0];
"Félix Brezo"
```
Regarding 2, there are only three places in which the author name appears in the HTML. The `<title>` together with some extra information, this `<meta>` and the already discarded `<strong class='display-name__html p-name emojify'>`. I don't really know where else to grab it from.

Reply:
> No, since it's a split and only the first part of the og:title tag is grabbed.

Sorry, I misread. A regexp that anchored the final `(` and `)` to the end would technically be better, since right now it would be wrong if someone included parentheses in their display name, though I don't know if that's possible on Mastodon. (Not a big deal regardless.)

> the already discarded `<strong class='display-name__html p-name emojify'>`

I didn't say not to use that — I said to only use one class, not all three, such that a simple change to any of them would break it. You [can do that with an XPath](https://stackoverflow.com/a/1604480), but it'd be better to add [these temporary lines](https://github.com/zotero/translators/blob/50aece7c7dfd4e6a57967c8aad8f50bd06c82e85/ACM%20Digital%20Library.js#L37-L39) to the top of the code and then use `text(.p-name)` (a CSS selector) to extract the author.

Reply:
I have just made an update since I noticed that the aforementioned approach was not correctly working when saving toots on threads. The aforementioned approach was assuming that the first toot was the one to quote, but it is not the case if the toot is part of a thread. I had to capture the highlighted publication using `//a[@class='detailed-status__display-name u-url']` and `//div[@class='detailed-status detailed-status--flex detailed-status-public']`. I know that there are too much classes to wrap but just choosing one of them (`//a[@class='detailed-status__display-name'` or  `//div[@class='detailed-status'`) was returning no results.

At the same time, I thought that I could use structures as `//a//strong` instead of specifying the full path `//a/span/bdi/strong`. I am finding out that I have more gaps on how Xpath works than I initially thought. Sorry for the inconvenience and thanks for the help anyway. Let's see if we can have this merged soon.

Reply:
I'm not going to approve this with such specific XPaths — there's just no reason for it. It's better to spend time understanding how to write resilient translators than to write a translator that will just break and take up people's time unnecessarily later.

I linked to instructions above for matching on specific classes with an XPath — just using `class=` will not work — but as I say, you should stop using XPaths and just use text() with CSS selectors instead.

Reply:
I have gone through this old translator which wasn't approved I have to admit that this is awkward for me since I cannot understand the function provided, whcih I have manually added at the beginning of my code as suggested:

```
// attr()/text() v2
// eslint-disable-next-line
function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
```
I have made it work for a simple example making these two lines match exactly:
```
		//newItem.title = ZU.xpathText(doc, "//meta[@name='description']/@content").replace(/\n/g, " ").replace(/ +/g, " ");
		newItem.title = attr(doc, 'meta[name=description]', 'content').replace(/\n/g, " ").replace(/ +/g, " ");
```
However, I'm struggling with more complex examples such as the following which dramatically crashes:
```
		//let tmpDate = ZU.xpathText(doc, "//div[@class='detailed-status detailed-status--flex detailed-status-public']/div/data/@value");
		let tmpDate = attr(doc, 'div[class=detailed-status detailed-status--flex detailed-status-public]/div/data', 'value');
```
Where can I find more documentation on how to use the provided function?

Reply:
`attr` and `text` use standard [CSS selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors), the same as you would use in a CSS file (or `document.querySelector()`).

E.g., you would match a class with a [class selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors) like `.detailed-status-public`.